check.go

```diff
@@ -5,6 +5,13 @@ import (
 )
 
 func (d *DBFT[H]) checkPrepare() {
+	if d.lastBlockIndex != d.BlockIndex || d.lastBlockView != d.ViewNumber {
+		// Notice that lastBlockTimestamp is left unchanged because
+		// this must be the value from the last header.
+		d.lastBlockTime = d.Timer.Now()
```

Comment:
With this change we need to adjust the comment of `lastBlockTime`:

> Wall clock time of when the last block was first seen (used for timer adjustments).

Because now it seems to me more like the time when proposal was firstly seen.